metapool/prep.py

```diff
@@ -0,0 +1,424 @@
+    str
+        Instrument model.
+    str
+        Run center based on the machine's id.
+    """
+    run_center = "UCSDMI"
+    instrument_model = instrument_code.split('_')[0]
+
+    if instrument_model in INSTRUMENT_LOOKUP.index:
+        run_center = INSTRUMENT_LOOKUP.loc[instrument_model, 'run_center']
+        instrument_model = INSTRUMENT_LOOKUP.loc[instrument_model, 'Vocab']
+    else:
+        instrument_prefix = get_machine_code(instrument_model)
+
+        if instrument_prefix not in INSTRUMENT_LOOKUP['machine prefix']:
+            ValueError('Unrecognized machine prefix %s' % instrument_prefix)
+
+        instrument_model = INSTRUMENT_LOOKUP[
+            INSTRUMENT_LOOKUP['machine prefix'] == instrument_prefix
+        ]['Vocab'].unique()[0]
+
+    return instrument_model, run_center
+
+
+def agp_transform(frame, study_id):
+    """If the prep belongs to the American Gut Project fill in some blanks
+
+    Parameters
+    ----------
+    frame: pd.DataFrame
+        The preparation file for a single project.
+    study_id: str
+        The Qiita study identifier for this preparations.
+
+    Returns
+    -------
+    pd.DataFrame:
+        If the study_id is "10317" then the `center_name`,
+        'library_construction_protocol', and `experiment_design_description`
+        columns are filled in with default values. Otherwise no changes are
+        made to `frame`.
+    """
+    if study_id == '10317':
+        frame['center_name'] = 'UCSDMI'
+        frame['library_construction_protocol'] = 'Knight Lab KHP'
+        frame['experiment_design_description'] = (
+            'samples of skin, saliva and feces and other samples from the AGP')
+    return frame
+
+
+def _check_invalid_names(sample_names):
+    # taken from qiita.qiita_db.metadata.util.get_invalid_sample_names
+    valid = set(ascii_letters + digits + '.')
+
+    def _has_invalid_chars(name):
+        return bool(set(name) - valid)
+
+    invalid = sample_names[sample_names.apply(_has_invalid_chars)]
+
+    if len(invalid):
+        warnings.warn('The following sample names have invalid '
+                      'characters: %s' %
+                      ', '.join(['"%s"' % i for i in invalid.values]))
+
+
+def preparations_for_run(run_path, sheet, pipeline=None):
+    """Given a run's path and sample sheet generates preparation files
+
+    Parameters
+    ----------
+    run_path: str
+        Path to the run folder
+    sheet: sample_sheet.SampleSheet
+        Sample sheet to convert
+    pipeline: str, optional
+        Which pipeline generated the data. The important difference is that
+        `atropos-and-bowtie2` saves intermediate files, whereas
+        `fastp-and-minimap2` doesn't. Defaults to `fastp-and-minimap2`, the
+        latest version of the sequence processing pipeline.
+
+    Returns
+    -------
+    dict
+        Filename to preparation file dictionary. Preparation files are
+        represented as DataFrames.
+    """
+
+    if pipeline is None:
```

Comment:
Maybe it would be better if the default for pipeline is 'fastp-and-minimap2', no?

Reply:
Good call, I think I was thinking of how lists/dicts can be modified when they are default values.